src/wavesurfer.ts

```diff
@@ -366,6 +372,15 @@ class WaveSurfer extends Player<WaveSurferEvents> {
     if (!blob && !channelData) {
       const onProgress = (percentage: number) => this.emit('loading', percentage)
       blob = await Fetcher.fetchBlob(url, onProgress, this.options.fetchParams)
+        .then((blob) => blob)
+        .catch(() => {
+          this.emit('error')
+          return undefined
```

Comment:
This would be a breaking change since it's swallowing the error that previously rejecting the promise returned from `wavesurfer.load`. Should be fine to release in 7.8.0 though.

Reply:
> The load function can also throw if decoding the audio has failed.

ah good to know

Reply:
Can you catch that error too?

Reply:
@katspaugh i think it is on line 243 

https://github.com/katspaugh/wavesurfer.js/pull/3532/files#diff-3e036de76e58315f9cb343d5cc0c2a80f8299050b8f6aff522b95e52429344c1R243

Reply:
I meant the peaks decoding [on line 402](https://github.com/katspaugh/wavesurfer.js/blob/e87542289d23d4cf243bff16ba828caf0a44ad50/src/wavesurfer.ts#L402).

Reply:
oooh i see, yeah i can add a `.catch` there too :)